components/AuthenticatedPage.js

```diff
@@ -58,7 +60,11 @@ class AuthenticatedPage extends React.Component {
                   defaultMessage="You need to be logged in to continue."
                 />
               </MessageBox>
-              <SignInOrJoinFree form="signin" disableSignup={this.props.disableSignup} />
+              <SignInOrJoinFree
+                form="signin"
+                routes={{ join: `/create-account?next=${encodeURIComponent(this.props.router.asPath)}` }}
```

Comment:
If `routes.join` is not provided, we're supposed to do a `switchForm` which  will display the signup form without changing the route. Could we rather fix this root issue?